presto-main/src/main/java/io/prestosql/sql/gen/BytecodeUtils.java

```diff
@@ -350,9 +345,6 @@ else if (type == ConnectorSession.class) {
         if (functionMetadata.isNullable()) {
             block.append(unboxPrimitiveIfNecessary(scope, returnType));
         }
-        if (Object.class.equals(returnType)) {
-            block.comment("Cast to proper SQL return type").checkCast(sqlReturnType.getJavaType());
```

Comment:
This got reverted. Was that an abandoned attempt approach?

Reply:
Yes; see my initial PR comment.